src/hooks/transactions/batch/useSendBatchTransactions.ts

```diff
@@ -0,0 +1,57 @@
+import { useCallback, useState } from 'react';
+import {
+  sendBatchTransactions,
+  SendBatchTransactionsPropsType
+} from 'services/transactions/sendBatchTransactions';
+import { BatchTransactionStatus } from 'types';
+import { removeSignedTransaction } from 'services';
+import { useDispatch } from 'reduxStore/DappProviderContext';
+import {
+  clearBatchTransactions,
+  setBatchTransactions
+} from 'reduxStore/slices';
+
+export const useSendBatchTransactions = () => {
+  const dispatch = useDispatch();
+  const [batchId, setBatchId] = useState<string>();
+
+  const send = useCallback(
+    async (params: SendBatchTransactionsPropsType) => {
+      const response = await sendBatchTransactions(params);
+      const error = response?.error;
+      const data = response?.data;
+
+      if (data) {
+        dispatch(setBatchTransactions(data));
+      }
+      setBatchId(data?.id);
```

Comment:
What if data is undefined?

Maybe better: if (!response?.data) return

Reply:
This situation is handled in the code below. If something went wrong then we log an error message plus some storage sanitization